### ISSUE TITLE
Add compute-envs wait subcommand

### ISSUE THREAD

**Issue by jordeu** [seqeralabs/tower-cli#12]
If the CLI is used in a script it is interesting to have a wait subcommand to wait for a compute environment available status.

Usage example:
```
towr compute-envs wait -i <compute_env_id> --status AVAILABLE --timeout 10m
towr compute-envs wait -n <compute_env_name> --no-status CREATING --timeout 10m
```

If the timeout is reached the process must return a non-zero exit code.

**Comment by jordeu** [comment]
An alternative to this subcommand, can be to add a `--wait-available` option at `tw compute-envs create ...` with an optional timeout using a [picocli](https://picocli.info/#fallbackValue-annotation) `fallbackValue`.

**Comment by pditommaso** [comment]
Think the wait as option makes more sense, but would not be better to have `--wait <STATUS>`, it could be more generic

**Comment by jordeu** [comment]
> Think the wait as option makes more sense, but would not be better to have `--wait <STATUS>`, it could be more generic

The problem with a generic `--wait` is that we need to fail on an error status and this depends on the context (workflow fail on "cancel" for example). Also with `--wait-available <timeout>` you have a way to define a timeout. What do we expect if the user says `--wait ERRORED` but this status is never reached?

So I like the idea of a more generic `--wait <STATUS>` but then I don't see how it will be clear when the command has to fail with an exit error code, and how we can change the default timeout.

**Comment by pditommaso** [comment]
I think it could be possible to validate the status provided is correct. Then the wait could have an optional timeout ie. `--wait <STATUS>[,TIMEOUT SECS]`

**Comment by drpatelh** [comment]
What about a sub-command that fetches the status directly and exits when it is met (and fails on an error)? I think having a `wait` sub-command would be confusing - it makes sense when waiting for a run to complete but it isn't immediately obvious why you would wait for a `compute-env`?

```
towr compute-envs status -i <compute_env_id> --timeout 10m AVAILABLE
```

I personally think it would be nicer to wait for a `compute-env` to be `CREATED`  by default before exiting when using the commands below. It saves the hassle of checking for this manually and it only takes a few seconds.

```
tw compute-envs add
tw compute-envs import
```

**Comment by jordeu** [comment]
@drpatelh I think that we agree that to add an option to the `add` and `import` it is a better than a new subcommand.

Last proposal is to have a  `--wait <STATUS>,[TIMEOUT SECS]` option the CLI will wait until <STATUS> and exit with a zero exit code. If the compute-env reaches a status that make it impossible to reach the proposed <STATUS> it will exit with a error one exit code.

**Comment by drpatelh** [comment]
> @drpatelh I think that we agree that to add an option to the add and import it is a better than a new subcommand.

Yup, but why have a `--wait` sub-command when you can wait by default for the `compute-env` to be created before exiting those commands? That would be the expected behaviour no?

**Comment by pditommaso** [comment]
To keep the `--wait` option semantic the same across resources ie. pipelines, compute env, actions, etc.

**Comment by ewels** [comment]
x-ref similar request for `tw launch`: https://github.com/seqeralabs/tower-cli/issues/177
